File: 14_Plus_One/solution.py
def plusOne( digits):
    """
    :type digits: List[int]
    :rtype: List[int]
    """
    num = 0
    c = 1
    for i in range(len(digits)-1,-1,-1):
        num += digits[i] * c 
        c = c * 10   
    num = num + 1
    return_list = [0] * len(str(num))
    iter = 1

    for i in range(len(str(num))-1,-1,-1):
       return_list[i] = (num //iter)%10
       iter = iter * 10

    return return_list


digits = [1,2,3,9]
print(plusOne( digits))

File: 14_Plus_One/solution.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> plusOne(vector<int>& digits) {
    int n = digits.size();
    for(int i = n-1; i >= 0 ; --i){
        if(digits[i] < 9){
            ++digits[i];
            return digits;
        }
        digits[i] = 0;
    }
    digits.insert(digits.begin(),1);
    return digits;
}

int main() {
    vector<int> digits = {1,2,3};
    vector<int> result = plusOne(digits);
    for(auto i : result) {
        cout << i << endl;
    }    
    return 0;
}